src/hooks/useLocalActiveCurrency.ts

```diff
@@ -0,0 +1,29 @@
+import { DEFAULT_LOCAL_CURRENCY, SUPPORTED_LOCAL_CURRENCIES, SupportedLocalCurrency } from 'constants/localCurrencies'
+import { atomWithStorage, useAtomValue } from 'jotai/utils'
+import { useMemo } from 'react'
+
+import useParsedQueryString from './useParsedQueryString'
+
+export const activeLocalCurrencyAtom = atomWithStorage<SupportedLocalCurrency>(
+  'activeLocalCurrency',
+  DEFAULT_LOCAL_CURRENCY
+)
+
+function useUrlLocalCurrency() {
+  const parsed = useParsedQueryString()
+  const parsedLocalCurrency = parsed.cur
+
+  if (typeof parsedLocalCurrency !== 'string') return undefined
+
+  const lowerCaseSupportedLocalCurrency = parsedLocalCurrency.toLowerCase()
+  return SUPPORTED_LOCAL_CURRENCIES.find(
+    (localCurrency) => localCurrency.toLowerCase() === lowerCaseSupportedLocalCurrency
+  )
+}
+
+export function useActiveLocalCurrency(): SupportedLocalCurrency {
```

Comment:
🚨 File name is different than hook name 🚔

Reply:
thats cringe